modules/cnf-reducing-netqueues-using-pao.adoc

```diff
@@ -0,0 +1,346 @@
+
+[id="adjusting-nic-queues-with-the-performance-profile_{context}"]
+== Adjusting the NIC queues with the performance profile
+
+The performance profile lets you adjust the queue count for each network device.
+
+Supported network devices are:
+
+* Non virtual network devices
+
+* Network devices that support multiple queues (channels)
+
+Unsupported network devices are:
+
+* Pure software network interfaces
+
+* Block devices
+
+* Intel DPDK virtual functions
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+* Install the OpenShift CLI (`oc`).
+
+
+.Procedure
+
+. Log in to the OpenShift Container Platform cluster running the Performance Addon Operator as a user with `cluster-admin` privileges.
+
+. Create a performance profile that is appropriate for your hardware and topology. For more information see the Tuning nodes for low latency with the performance profile section. Alternatively, edit an existing performance profile using the following command:
++
+[source,terminal]
+----
+$ oc edit performanceprofile <your_profile_name>
+----
+
+. Populate the `spec` field profile with the `net` object. The object list can contain two fields:
+
+* `userLevelNetworking` is a required field that specifies a boolean flag which, if `true` sets the queue count to the reserved CPU count for all supported devices. The default is `false`.
+* `Devices` is an optional field which specifies a list of devices that will have the queues set to the reserved CPU count. If the device list is empty the configuration applies to all network devices. The configuration is as follows:
+** `interfaceName`: Name supports shell-style wildcards which can be positive or negative.
+*** Example wildcard syntax is as follows: `<string> .*`
+*** Negative rules are prefixed with an exclamation mark. To apply the net queue changes to all devices other than the excluded list, use  `!<device>`; for example, `!eno1`.
+* `vendorID`: Network device vendor ID represented as a 16 bit hexmadecimal number with a 0x prefix.
+* `deviceID`: Network device ID (model) represented as a 16 bit hexmadecimal number with a 0x prefix.
++
+[NOTE]
+====
+When a `deviceID` is specified, the `vendorID` must also be defined. A device that matches all of the device identifiers specified in a device entry `interfaceName`, `vendorID` or a pair of `vendorID` plus `deviceID` representing a network device qualifies as a network device that will have its net queues count set to the reserved CPU count.
+
+When two or more devices are specified, the net queues count is set to any net device that matches one of them.
+====
++
+An example performance profile configuration is shown below:
++
+[source,yaml]
+----
+apiVersion: performance.openshift.io/v2
+  kind: PerformanceProfile
+  metadata:
+    name: manual
+  spec:
+    cpu:
+    isolated: 3-51,54-103
+    reserved: 0-2,52-54
+
+    net:
+    - userLevelNetworking: true         <1>
+    #!
+
+    more examples:
+
+      net:
```

Comment:
All of these additional `net` sections are at different indentation levels. I'd make sure they are at the correct level at least.